src/AutoRest.CSharp/Common/Output/Models/Types/ExpressionTypeProvider.cs

```diff
@@ -12,29 +13,52 @@ namespace AutoRest.CSharp.Output.Models.Types
     // TODO -- eventually we should combine everything in this class into TypeProvider
     internal abstract class ExpressionTypeProvider : TypeProvider
     {
-        protected ExpressionTypeProvider(string defaultNamespace, SourceInputModel? sourceInputModel) : base(defaultNamespace, sourceInputModel)
+        protected ExpressionTypeProvider(string defaultNamespace, SourceInputModel? sourceInputModel)
+            : base(defaultNamespace, sourceInputModel)
         {
+            DeclarationModifiers = ClassSignatureModifiers.Partial | ClassSignatureModifiers.Public;
```

Comment:
I like this idea, but I think we should call this `TypeDeclarationModifier` or `TypeSignatureModifier` since it might apply to struct and enum as well

Reply:
Issue here: https://github.com/Azure/autorest.csharp/issues/4311